app/models/manageiq/providers/foreman/provider.rb

```diff
@@ -126,9 +126,11 @@ def verify_credentials(auth_type = nil, options = {})
   def ensure_managers
     build_provisioning_manager unless provisioning_manager
     provisioning_manager.name    = "#{name} Provisioning Manager"
+    provisioning_manager.provider = self
 
     build_configuration_manager unless configuration_manager
     configuration_manager.name    = "#{name} Configuration Manager"
```

Comment:
This would probably need a safeguard to prevent appending the suffix more than once:
```suggestion
    configuration_manager.name    = "#{name} Configuration Manager" unless name.end_with?(' Configuration Manager')
```